src/core/lib/iomgr/tcp_uv.cc

```diff
@@ -356,7 +356,9 @@ static grpc_resolved_addresses* handle_addrinfo_result(
     i++;
     prev = resp;
     resp = resp->ai_next;
-    gpr_free(prev);
+    // addrinfo objects are allocated by libuv (e.g. in uv_getaddrinfo)
+    // and not by gpr_malloc
+    free(prev);
```

Comment:
Please doublecheck, I'm not familiar with the libuv code.

Reply:
If this is uv allocated memory, we might want to go all the way and just skip the free here and at the end of the function
call ```uv_freeaddrinfo(result)```